src/model/UInt64.ts

```diff
@@ -37,6 +37,9 @@ export class UInt64 {
      * @returns {UInt64}
      */
     public static fromUint(value: number): UInt64 {
+        if (value < 0) {
```

Comment:
This check would better be done in the class constructor.

Also the error message contains a typo. Either of 'UInt' or 'Unsigned Integer' would be fine 👍

Reply:
@evias 32bit unsigned int capped at 4294967295 and -1 >>> 0 === 4294967295. Constructor only takes UInt arrays, how does it differentiate -1 and real 4294967295?

Reply:
I agree @rg911, you are right. 
I believe adding it in fromUInt makes sense like this then :+1:

Reply:
👍

Reply:
More clean way is to add the check in NEM2-library I'd thought?